### ISSUE TITLE
Doesn't show databases, tables etc on some servers

### ISSUE THREAD

**Issue by DennisJakobsen** [microsoft/azuredatastudio#12992]
- Azure Data Studio Version:
```
Version: 1.23.0
Commit: d296b6397e0acfddc57e9085e736e084969cdaeb
Date: 2020-10-13T22:58:36.079Z
VS Code: 1.49.0
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0
```

Since upgrading to 1.23.0 (it worked in 1.22.1) I'm unable to see the databases, tables, or anything when connecting to some SQL servers. Screenshot:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/3790501/96493317-1fc54f00-1245-11eb-9463-5f352a462345.png">

It works fine for the server I'm running in docker, but not the other one. I'm still able to execute queries on the server. 

In `renderer1.log` I get the following error:
```
[2020-10-19 19:49:27.174] [renderer1] [error] An unknown error occurred. Please consult the log for more details.
[2020-10-19 19:49:27.175] [renderer1] [error] Cannot read property 'isCloud' of null: TypeError: Cannot read property 'isCloud' of null
	at t.MssqlIconProvider.getConnectionIconId (/Applications/Azure Data Studio.app/Contents/Resources/app/extensions/mssql/dist/main.js:411:37866)
	at d.$getConnectionIconId (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:697:248)
	at g._doInvokeHandler (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:969:707)
	at g._invokeHandler (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:969:399)
	at g._receiveRequest (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:968:59)
	at g._receiveOneMessage (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:966:886)
	at /Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:965:34
	at l.fire (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:48:475)
	at v.fire (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:259:381)
	at /Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:1149:963
	at l.fire (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:48:475)
	at v.fire (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:259:381)
	at t.PersistentProtocol._receiveMessage (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:264:451)
	at /Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:261:489
	at l.fire (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:48:475)
	at p.acceptChunk (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:256:851)
	at /Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:256:203
	at Socket.t (/Applications/Azure Data Studio.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:266:54)
	at Socket.emit (events.js:223:5)
	at addChunk (_stream_readable.js:309:12)
	at readableAddChunk (_stream_readable.js:290:11)
	at Socket.Readable.push (_stream_readable.js:224:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
```

I have had someone else test it as well with the same result

**Comment by david-driscoll** [comment]
I'm also seeing this issue as well

**Comment by Charles-Gagnon** [comment]
@david-driscoll @DennisJakobsen What type of server are you connecting to that you see this? @david-driscoll - do you also only see this for some servers?

**Comment by david-driscoll** [comment]
In my case it's an on-premise SQL Server server, I can't recall the version but I think we're actually on 2019 now.

**Comment by david-driscoll** [comment]
Oh and rolling back manually to 1.22.1 did fix the problem for me.

**Comment by Charles-Gagnon** [comment]
Getting some logs here might help - to find your logs:

Open command palette (Click View -> Command Palette)
Run the command: Developer: Open Logs Folder

This will open the log file locally. Please include renderer.log

**Comment by Charles-Gagnon** [comment]
In addition - please follow these instructions and get an additional log file : 

1. Open command palette
1. Run the command `Developer: Open Extension Logs Folder`
1. Navigate to the `Microsoft.mssql` folder

There should be a `sqltools.log` file there - either attach it to a response or e-mail it directly to me.

**Comment by covlorin** [comment]
I'm seeing the same issue and error message in the renderer1 file as reported above, ie `TypeError: Cannot read property 'isCloud' of null` . 

ADS Version 1.23.0 
FAILS on: Microsoft SQL Server Enterprise (64-bit) Version 15.0.2070.41 on Windows Server 2019 Standard (10.0) 
WORKS on: Microsoft SQL Server Enterprise (64-bit) Version 13.0.5622.0 on Microsoft Windows NT 6.3 (9600) 

Note, the `Microsoft.mssql` folder described above is empty.

**Comment by DennisJakobsen** [comment]
I see it failing on `Microsoft SQL Server 2019 (RTM-CU1) (KB4527376) - 15.0.4003.23 (X64) 	Dec  6 2019 14:53:33 	Copyright (C) 2019 Microsoft Corporation	Web Edition (64-bit) on Windows Server 2019 Standard 10.0 <X64> (Build 17763: ) (Hypervisor)`

My `Microsoft.mssql` folder is empty

**Comment by Charles-Gagnon** [comment]
For those who don't have a sqltools.log - could you go into settings and verify you have the following settings set to the values you see here : 

![image](https://user-images.githubusercontent.com/28519865/97196415-adcf9700-1769-11eb-95fb-be5342765631.png)

![image](https://user-images.githubusercontent.com/28519865/97196494-c93aa200-1769-11eb-9ea1-52e21d09f81b.png)

And then also make sure you have this setting disabled : 

![image](https://user-images.githubusercontent.com/28519865/97196664-f6875000-1769-11eb-9002-2e3d1da9fae7.png)


In addition - if someone could try out the latest insiders build (https://github.com/microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-main) and see if they still have the same issue that'd be great. If you do please get the two logs (or 1 if sqltools.log doesn't exist)

**Comment by linhoz** [comment]
Hi.

Installing the Insiders Build I have the same issue and the Server Tree_
Use Async Server Tree is unchecked.

I checked all option like you said and my Microsoft.mssql folder is empty.

Tks.
JS

Em seg., 26 de out. de 2020 às 16:03, Charles Gagnon <
notifications@github.com> escreveu:

**Comment by Charles-Gagnon** [comment]
Can you check the render1.log to see if you have the `Cannot read property 'isCloud' of null: TypeError: Cannot read property 'isCloud' of null` message in there? (make sure you use the command from the insiders version of ADS so you open the correct folder)

If there are any other errors there too please include them.

**Comment by hfleitas** [comment]
I installed CU8 for SQL Server 2019 and this seemed to have made a difference. 

The problem seems to be occurring specifically with with  ADS version:

Version: 1.23.0 (user setup)
Commit: d296b6397e0acfddc57e9085e736e084969cdaeb
Date: 2020-10-13T23:03:09.799Z
VS Code: 1.49.0
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19041

When connecting to SQL Server 2019 15.0.4033.1 (CU4).

**Comment by DennisJakobsen** [comment]
sqltools.log:
```
20-10-26 17:28:07.7488070 pid:79062 tid:1 sqltools Information: 0 : Initialized the sqltools logger. Log file is: /Users/dennisjakobsen/Library/Application Support/azuredatastudio/logs/20201026T172805/exthost1/Microsoft.mssql/sqltools.log
20-10-26 17:28:08.5766340 pid:79062 tid:1 sqltools Verbose: 0 : TaskService initialized
20-10-26 17:28:08.5794760 pid:79062 tid:1 sqltools Verbose: 0 : SerializationService initialized
20-10-26 17:28:08.5803110 pid:79062 tid:1 sqltools Verbose: 0 : ObjectExplorer service initialized
20-10-26 17:28:08.5823260 pid:79062 tid:1 sqltools Verbose: 0 : TSqlFormatter initialized
20-10-26 17:28:08.6113980 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[0], of type[Request] and method[initialize]
20-10-26 17:28:08.6196930 pid:79062 tid:10 sqltools Verbose: 0 : Initializing workspace service
20-10-26 17:28:08.6579820 pid:79062 tid:10 sqltools Verbose: 0 : Sending message of id[0], of type[Response] and method[initialize]
20-10-26 17:28:08.6739330 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[], of type[Event] and method[initialized]
20-10-26 17:28:08.6741730 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[], of type[Event] and method[workspace/didChangeConfiguration]
20-10-26 17:28:08.7023420 pid:79062 tid:10 sqltools Verbose: 0 : HandleDidChangeConfigurationNotification
20-10-26 17:28:08.7132460 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[1], of type[Request] and method[capabilities/list]
20-10-26 17:28:08.7428260 pid:79062 tid:10 sqltools Verbose: 0 : Sending message of id[1], of type[Response] and method[capabilities/list]
20-10-26 17:28:08.8700190 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[2], of type[Request] and method[connection/connect]
20-10-26 17:28:08.8771690 pid:79062 tid:10 sqltools Verbose: 0 : HandleConnectRequest
20-10-26 17:28:08.8791250 pid:79062 tid:10 sqltools Verbose: 0 : Sending message of id[2], of type[Response] and method[connection/connect]
20-10-26 17:28:08.8828370 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[], of type[Event] and method[connection/languageflavorchanged]
20-10-26 17:28:08.8885990 pid:79062 tid:10 sqltools Verbose: 0 : HandleRebuildIntelliSenseNotification
20-10-26 17:28:09.5330830 pid:79062 tid:10 sqltools Verbose: 0 : Sending message of id[], of type[Event] and method[connection/complete]
20-10-26 17:28:09.7555730 pid:79062 tid:13 MicrosoftSqlToolsServiceLayer Information: 0 : 2020-10-26T17:28:09.7550800+01:00 - get data for urn: Server[@Name='mssql4']
20-10-26 17:28:09.8306850 pid:79062 tid:10 sqltools Verbose: 0 : Received message with id[3], of type[Request] and method[objectexplorer/createsession]
20-10-26 17:28:09.8317320 pid:79062 tid:10 sqltools Verbose: 0 : HandleCreateSessionRequest
20-10-26 17:28:09.8330920 pid:79062 tid:10 sqltools Verbose: 0 : HandleCreateSessionRequest
20-10-26 17:28:09.8385140 pid:79062 tid:10 sqltools Verbose: 0 : Sending message of id[3], of type[Response] and method[objectexplorer/createsession]
20-10-26 17:28:09.8397400 pid:79062 tid:10 sqltools Information: 0 : Creating OE session
20-10-26 17:28:09.9043280 pid:79062 tid:13 MicrosoftSqlToolsServiceLayer Information: 0 : 2020-10-26T17:28:09.9042960+01:00 - Missing regular property Edition property bag state Empty for type Information
20-10-26 17:28:09.9066760 pid:79062 tid:13 MicrosoftSqlToolsServiceLayer Information: 0 : 2020-10-26T17:28:09.9066660+01:00 - get data for urn: Server[@Name='mssql4']/Information
20-10-26 17:28:09.9810900 pid:79062 tid:13 MicrosoftSqlToolsServiceLayer Information: 0 : 2020-10-26T17:28:09.9810710+01:00 - get data for urn: Server/Database[@Name='master']
20-10-26 17:28:10.1505670 pid:79062 tid:15 sqltools Error: 0 : Unexpected exception on the binding queue: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.SqlTools.ServiceLayer.ObjectExplorer.ServerVersionHelper.CalculateServerType(ServerInfo serverInfo) in D:\a\1\s\src\Microsoft.SqlTools.ServiceLayer\ObjectExplorer\SqlServerType.cs:
```

Async server tree is unchecked

**Comment by linhoz** [comment]
See the log

Em seg., 26 de out. de 2020 às 16:28, Charles Gagnon <
notifications@github.com> escreveu:

**Comment by Charles-Gagnon** [comment]
Something else to try and track down where the issue is happening - could someone with the issue try going to the install folder for ADS (Windows User Install - `%LOCALAPPDATA%\Programs\Azure Data Studio\resources\app\extensions\mssql`) and opening the `config.json` file there - then updating this line 

`"version": "3.0.0-release.38",`

to be version .35 instead 

`"version": "3.0.0-release.35",`

And restarting ADS, then seeing if the issue still persists?

**Note - even if this works this is not a suggested workaround as there could be unknown side effects. After testing to see if the issue is resolved it would be best to revert back to .38 until an actual fix can be found** (in the meantime using the [1.22.1](https://github.com/microsoft/azuredatastudio/releases/tag/1.22.1) version should unblock you)

**Comment by covlorin** [comment]
For me the config is found in `C:\Program Files\Azure Data Studio\resources\app\extensions\mssql` . Changing the version string and relaunching results in error on launch: Failed to start SQL Tools Service. After that can't open any servers. 

> Something else to try and track down where the issue is happening - could someone with the issue try going to the install folder for ADS (Windows User Install - `%LOCALAPPDATA%\Programs\Azure Data Studio\resources\app\extensions\mssql`) and opening the `config.json` file there - then updating this line
> 
> `"version": "3.0.0-release.38",`
> 
> to be version .35 instead
> 
> `"version": "3.0.0-release.35",`
> 
> And restarting ADS, then seeing if the issue still persists?
>

**Comment by DennisJakobsen** [comment]
> Something else to try and track down where the issue is happening - could someone with the issue try going to the install folder for ADS (Windows User Install - `%LOCALAPPDATA%\Programs\Azure Data Studio\resources\app\extensions\mssql`) and opening the `config.json` file there - then updating this line
> 
> `"version": "3.0.0-release.38",`
> 
> to be version .35 instead
> 
> `"version": "3.0.0-release.35",`
> 
> And restarting ADS, then seeing if the issue still persists?
> 
> **Note - even if this works this is not a suggested workaround as there could be unknown side effects. After testing to see if the issue is resolved it would be best to revert back to .38 until an actual fix can be found** (in the meantime using the [1.22.1](https://github.com/microsoft/azuredatastudio/releases/tag/1.22.1) version should unblock you)

For me on Mac this "fixed" the problem

**Comment by linhoz** [comment]
For me this workaround works. Tks.

Em seg., 26 de out. de 2020 às 23:51, Charles Gagnon <
notifications@github.com> escreveu:

**Comment by Charles-Gagnon** [comment]
This might have something to do with this.

**Comment by linhoz** [comment]
Hi. Even after de new release 1.24 the problems continue.

Em seg., 23 de nov. de 2020 às 16:36, Charles Gagnon <
notifications@github.com> escreveu:

**Comment by Orionsgate** [comment]
Couple of things if it helps:

- This happens when logging in via SQL login. Using Windows Authentication, it works normally.
- When using SQL login, I also get `ERR Error rendering widget explorer-widget TypeError: Cannot read property 'isCloud' of null`

**Comment by eat-sleep-code** [comment]
This issue still occurs on current Insider Build on a Mac running Big Sur.

**Comment by hastiAu** [comment]
I cant connect to server please see my log file
[sqltools.log](https://github.com/microsoft/azuredatastudio/files/5798360/sqltools.log)

**Comment by hastiAu** [comment]
I finally can connect to server with Navicat 
but I cant connect with ADS
I do all of these ways but I can connect to server but cannot see any database.

It relates to permission on server? Or should i do any things?

<img width="380" alt="Screen Shot 2021-01-17 at 4 33 27 PM" src="https://user-images.githubusercontent.com/67632833/104843799-c4593a80-58e1-11eb-843a-f6941bd94d05.png">

**Comment by gjh71** [comment]
Like mentioned by hfleitas (https://github.com/microsoft/azuredatastudio/issues/12992#issuecomment-716667849) the problem was solved in my case, by upgrading the sql-server 2019 server with CU-8.
I've noticed that I was able to get the database list with 'sysadmin' rights, but I preferred applying the cumulative update :-)

> I finally can connect to server with Navicat
> but I cant connect with ADS
> I do all of these ways but I can connect to server but cannot see any database.
> 
> It relates to permission on server? Or should i do any things?
> 
> <img alt="Screen Shot 2021-01-17 at 4 33 27 PM" width="380" src="https://user-images.githubusercontent.com/67632833/104843799-c4593a80-58e1-11eb-843a-f6941bd94d05.png">

**Comment by covlorin** [comment]
> Like mentioned by hfleitas ([#12992 (comment)](https://github.com/microsoft/azuredatastudio/issues/12992#issuecomment-716667849)) the problem was solved in my case, by upgrading the sql-server 2019 server with CU-8.
> I've noticed that I was able to get the database list with 'sysadmin' rights, but I preferred applying the cumulative update :-)
> 

Glad applying an update worked for you, but it doesn't work for us. 
The servers we are having this problem with are SQL Server 2019 that are already update with a higher version than CU-8 OR KB4577194. The version of the SQL Server update that is installed is GDR 15.0.2080.9(15.0.4083.0) with a KBKB4583458.

**Comment by rgreig** [comment]
This issue impacts me, but I have also found that I cannot view query plans either. With 1.22.1 or earlier everything is fine, but with 1.23.0 or later I have the issue of the server tree not populating and also being unable to view a query plan (i.e when I click Explain the query just executes but no plan is shown, and no tab is shown for the plan).

I don't see a specific error relating to the plan in the logs but I do have a warning when connecting:

[2021-02-11 17:14:39.970] [renderer1] [warning] Could not find ServerInfo for connection c89f3ee7-76b8-4a73-aba0-2cdf88546c8b when updating icon

Is this happening for anyone else who has reported this issue or is it an unrelated issue? It pretty much makes this version unusable for me so I am having to roll back to 1.22.1.

**Comment by leecheeaun** [comment]
Still experiencing this issue - connecting using Microsoft SQL Server with Auth type SQL Login -> Cant see any databases in the side bar after connecting. Queries run fine, but none of the DB > Tables / Views / Functions are populated in the sidebar. Surprisingly, the Postgres extension works flawlessly ...

**Comment by aaronostlund** [comment]
This is happening to me as well. It happens on windows auth.

**Comment by kburtram** [comment]
This looks similar to https://github.com/microsoft/azuredatastudio/issues/6450.  I close this as duplicate of that issue to consolidate the bug reports and diagnostic info.

**Comment by acohenOT** [comment]
@kburtram not sure these are duplicates. This problem started with 1.23.0 (it worked in 1.22.1) -- the other issue #6450 was a problem way before that. Also this issue has no timeout or any other error when you expand the Database Folder - it just shows nothing. Appears to only affect on-prem SQL Server and not Azure.

**Comment by LtFarns** [comment]
I am seeing this same issue with server connection on build 1.41.1. Can we get this item re opened?
